src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/TimeSpanConverter.cs

```diff
@@ -0,0 +1,54 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers.Text;
+using System.Diagnostics;
+
+namespace System.Text.Json.Serialization.Converters
+{
+    internal sealed class TimeSpanConverter : JsonConverter<TimeSpan>
+    {
+        public override TimeSpan Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            if (reader.TokenType != JsonTokenType.String)
+            {
+                throw ThrowHelper.GetInvalidOperationException_ExpectedString(reader.TokenType);
+            }
+
+            var source = reader.GetSpan();
```

Comment:
We need to also support escaped characters. See `DateTime` parsing logic as an [example](https://github.com/dotnet/runtime/blob/24950a310bde0d816060f920090b4e9666c15fed/src/libraries/System.Text.Json/src/System/Text/Json/Reader/JsonReaderHelper.cs#L254-L261).

Reply:
@layomia Can you give me a little more direction on this? How do I model a test case where `JsonConverter<T>.Read` would be fed an escaped string? I was thinking we would only run into this case supporting something like `Utf8JsonReader.Try/GetTimeSpan`.

Reply:
The bytes we are parsing - in `Utf8JsonReader.ValueSpan` and `Utf8JsonReader.ValueSpan` - are "raw", meaning that they contain the actual data in the input payload, which could be escaped.

Here are some test examples for `DateTime` - [here](https://github.com/dotnet/runtime/blob/0416c3469b7bccff087b89db3d9967dfd36070a3/src/libraries/System.Text.Json/tests/System.Text.Json.Tests/JsonDateTimeTestData.cs#L81-L85) and [here](https://github.com/dotnet/runtime/blob/0416c3469b7bccff087b89db3d9967dfd36070a3/src/libraries/System.Text.Json/tests/System.Text.Json.Tests/JsonDateTimeTestData.cs#L219-L230).

wrt this PR - you could just add some escaped variations of your existing tests (I usually find tables like this useful https://www.utf8-chartable.de/):

https://github.com/dotnet/runtime/blob/0293935ecbbf646bba68ce29ef57a4ced3f7f9c9/src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/Value.ReadTests.cs#L488-L502

Reply:
Thanks @layomia I will push an update shortly.

Reply:
Updated